Fix: Type search field overflow

rschristian opened the pull request:
### What kind of change does this PR introduce?
A fix to a bug

### Did you add tests for your changes?
No

### Summary
The input length for the types search field in the types prompt was not ever checked. So the following, when `.len()` returned 7+, was quite insane. 

```
let underscores = "_".repeat(6 - self.input.len());
```

Quite frankly I'm in awe of the output. Unless I've copy pasted incorrectly, Rust is trying to allocate 18,446 Petabytes worth of underscores. *Nice*

![temp](https://user-images.githubusercontent.com/33403762/95828192-48f05780-0cfa-11eb-9842-019543140584.png)

Of course this breaks terminal formatting as well as it's a hard crash of the program from within the `.with_raw()`. I have to hope that output number is some sort of UI glitch or I really have to ask the Rust team why that's not capped lower.

Edit: Ah, didn't realize `.repeat()` took a `u64` of all things. Thought it would be much more limited than that.

rschristian commented on src/prompt/type_prompt.rs:
```diff
@@ -121,12 +123,12 @@ impl<'a> TypePrompt<'a> {
                 style(s).with(Color::Magenta).to_string()
             });
 
-            let y_offset = header.len() as u16 + 1;
-
             for line in header {
                 buffer.push_line(line);
             }
 
+            let y_offset = buffer.lines() + 1;
+
```
This is just a consistency thing. `y_offset` is determined by `buffer.lines()` on the files prompt and it needs to be as it's the header + a few lines. I figure there's no reason not to do the same here. I don't think it's any slower and consistency is always nice.

brigand commented:
Yeah, so this is the u64 wrapping (which only panics in debug builds).

Interesting choice preventing more than 6 characters being entered instead of resolving the wrapping issue. I think this is fine since we never enter more than 6 characters, but it's still a potential future issue to have the wrapping be possible if we relax this restriction later.

I would also change the `repeat` subtraction to a `saturating_sub` to be future-safe.

rschristian commented:
Ah, fair enough. Didn't know Rust caught underflows in debug. Pretty neat. 

I capped at 6 as I assumed the filter would stop working after 6 characters, so any uncapped entry, even with proper overflow blocking, would be a waste. This obviously isn't the case, don't know why I assumed this. 

Anyways, I've reverted that, replaced with a saturating sub as you suggested.
